components/org.wso2.carbon.identity.application.authenticator.hypr.common/src/main/java/org/wso2/carbon/identity/application/authenticator/hypr/common/model/DeviceAuthenticationRequest.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright (c) 2023, WSO2 LLC. (http://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.application.authenticator.hypr.common.model;
+
+import org.wso2.carbon.identity.application.authenticator.hypr.common.web.HYPRWebUtils;
+
+import java.security.NoSuchAlgorithmException;
+import java.util.Map;
+
+/**
+ * Model class for the request which initiates the device authentication.
+ */
+public class DeviceAuthenticationRequest {
+
+    private final String namedUser;
+    private final String machine;
+    private final String machineId;
+    private final String appId;
+    private final String sessionNonce;
+    private final String deviceNonce;
+    private final String serviceNonce;
+    private final String serviceHmac;
+    private Map additionalDetails;
+
+    public DeviceAuthenticationRequest(String namedUser, String machineId, String appId)
+            throws NoSuchAlgorithmException {
+
+        this.namedUser = namedUser;
+        this.machineId = machineId;
+        this.appId = appId;
+        this.machine = "WEB";
```

Comment:
Shall we get it from a constant?

Reply:
Fixed with [4aa25f7](https://github.com/wso2-extensions/identity-outbound-auth-hypr/pull/1/commits/4aa25f7689bb9972bb19d4b147e15964fa127240)